src/odemis/driver/kryoz.py

```diff
@@ -0,0 +1,159 @@
+# -*- coding: utf-8 -*-
+'''
+Created on 5 November 2020
+
+@author: Anders Muskens
+
+Copyright © 2020 Anders Muskens, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License version 2 as published by the Free Software Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with Odemis. If not, see http://www.gnu.org/licenses/.
+'''
+
+from __future__ import division
+
+import logging
+
+from odemis import model
+from odemis.util import to_str_escape
+
+import threading
+import socket
+
+DEFAULT_PORT = 5041
+
+
+class CoolerBasic():
+    '''
+    Basic cooler driver
+    '''
+
+    def __init__(self, name, role, host, port=DEFAULT_PORT, daemon=None, **kwargs):
+        """
+        host: (string) the TCP/IP hostname of the server
+        port: (int) the TCP/IP port of the server.
+
+        Raises:
+            ValueError if no scanner child is present
+        """
+        #super(Cooler, self).__init__(name, role, daemon=daemon, **kwargs)
+
+        self._host = host
+        self._port = port
+        self._is_connected = False
+
+        try:
+            logging.debug("Connecting to %s:%d", self._host, self._port)
+            self._socket = socket.create_connection((self._host, self._port))
+            self._socket.settimeout(2.0)
+        except socket.error:
+            raise model.HwError("Failed to connect to '%s:%d', check that the Kryoz Cooler "
+                                "Server is operating, and connected to the network, turned "
+                                "on, and correctly configured." % (host, port))
+
+        # to acquire before sending anything on the socket
+        self._net_access = threading.Lock()
+
+    def terminate(self):
+        self._socket.shutdown(socket.SHUT_RDWR)
+        self._socket.close()
+
+        super(CoolerBasic, self).terminate()
+
+    def _sendQuery(self, cmd):
+        """
+        cmd (byte str): command to be sent to device
+        returns (byte str): answer received from the device
```

Comment:
Usually in such code we return unicode strings (as long as the return value is always a english-like text). That's simpler as otherwise every caller has to convert to unicode.